File: src/main/java/gangofthree/auth/service/AuthService.java
package gangofthree.auth.service;

import gangofthree.auth.dto.request.ForgotPasswordRequest;
import gangofthree.auth.dto.request.LoginRequest;
import gangofthree.auth.dto.request.OtpRequest;
import gangofthree.auth.dto.request.RegisterRequest;
import gangofthree.auth.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse login(LoginRequest loginRequest);
    AuthResponse verifyOtp(OtpRequest otpRequest);
    AuthResponse register(RegisterRequest registerRequest);
    AuthResponse forgotPassword(ForgotPasswordRequest forgotPasswordRequest);
}

File: frontend/src/features/auth/services/auth.service.ts
// src/features/auth/services/auth.service.ts
import apiClient from "@/services/client";

export const authService = {
    register: async (payload: any) => {
        const requestBody = {
            firstName: payload.firstName,
            lastName: payload.lastName,
            phoneNumber: payload.phone,
            email: payload.email,
            password: payload.password,
            confirmPassword: payload.confirmPassword,
            role: "BUYER",
            cityId: Number(payload.city_id),
        };

        const response = await apiClient.post("/auth/register", requestBody);
        return response.data;
    },

    loginWithPassword: async (identifier: string, pass: string) => {
        const response = await apiClient.post("/auth/login", {
            credential: identifier,
            password: pass
        });
        return response.data;
    },

    requestOtp: async (identifier: string) => {
        const response = await apiClient.post("/auth/login/otp-request", {
            credential: identifier
        });
        return response.data;
    },

    verifyOtp: async (identifier: string, otp: string) => {
        const response = await apiClient.post("/auth/verify-otp", {
            credential: identifier,
            otp: otp,
            purpose: "LOGIN"
        });
        return response.data;
    }
};